lib/src/git_backend.rs

```diff
@@ -557,6 +674,43 @@ fn prevent_gc(git_repo: &gix::Repository, id: &CommitId) -> Result<(), BackendEr
     Ok(())
 }
 
+/// This is an inlined part of gix::Repository::commit_as_inner (called from
+/// commit_as). It's different from prevent_gc because we update the reflog
+/// correctly as we're (should be) making a real commit here.
+fn prevent_gc_commit(
+    git_repo: &gix::Repository,
+    git_id: gix::Id,
+    commit: &gix::objs::Commit,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    git_repo.edit_reference(RefEdit {
+        change: Change::Update {
+            log: LogChange {
+                mode: RefLog::AndReference,
+                force_create_reflog: false,
+                message: gix::reference::log::message(
+                    "commit",
+                    commit.message.as_ref(),
+                    commit.parents.len(),
+                ),
+            },
+            expected: match commit.parents.first().map(|p| Target::Peeled(*p)) {
+                Some(previous) => {
+                    // if reference == "HEAD" { // no_gc_ref is never HEAD
+                    //     PreviousValue::MustExistAndMatch(previous)
+                    // } else {
+                    PreviousValue::ExistingMustMatch(previous)
```

Comment:
Since this is just a random ref preventing GC, there should be no previous ref. So I think this function is practically the same as `prevent_gc()`, but with random value and `MustNotExist`. The logic copied from gix is only needed for user-facing refs, I think.

BTW, as we now have the id of the commit object, it's probably better to use the commit id instead of random value.

Reply:
Yeah that's the part I wasn't too sure about.
We do create a reflog entry (afaiu) here and we do that just as we're creating a new commit - makes sense to me, and inlined code from gix to make sure the behaviour doesn't change (besides not creating said ref for intermediate commits in the race-preventing loop)

Reply:
This can be addressed as a follow up. I think it's also good to start with a simplified version of this function (as we won't need the full gix logic.)

Reply:
I was afraid to change the existing behavior by not inlining what gix did precisely, but if you're both certain then sure, I don't like that giant method that kind of does the same thing as existing `prevent_gc` myself :)

Reply:
> and `prevent_gc` absolutely does just that, it's just that it wasn't used in `write_commit`

Yep, that's just because `prevent_gc()` was added later (and gix `commit_as()` function requires some ref to move.) `prevent_gc()` would do what we want.

Reply:
I'm not sure I followed the conclusion of this thread. It sounded to me like we all agreed that using `prevent_gc()` instead of this function should work. Did I misunderstand? Oh, you did it in the last commit. Move it to this commit?

Reply:
Oh, right, I just amended review fixes, lol
One moment :)